### ISSUE TITLE
Add Mode Of Inheritance to Location Search

### ISSUE THREAD

**Issue by david-ma** [broadinstitute/seqr#2770]
Our users would like to filter the variant search based on Panel App Mode Of Inheritance (MOI).

E.g. only look for variants in genes that are known to be Autosomal Dominant, or Autosomal Recessive.

This could be done at the Location search component like so:
<img width="1041" alt="Screen Shot 2022-06-01 at 12 49 05 pm" src="https://user-images.githubusercontent.com/3320713/171649260-11cc7773-4a71-4c21-88c4-5cb34c00385a.png">

It would be too complex to give each gene a hover-over that shows all the PanelApp info on that gene, but by giving the user pre-checked checkboxes, which hide/show the genes instantly, it could be intuitive to the user which genes have which properties. And the help tooltip could explain MOI and/or link to the PanelApp User Guide.

I think that since the Location component has PanelApp info on the Gene List at this point in the user flow, (when you choose a PanelApp Gene List), it would be easier to filter the genes that are passed to Elasticsearch, than to add that logic to Elasticsearch.

**Comment by hanars** [comment]
I agree that doing the gene filtering in the UI is a better approach than putting the logic in elasticsearch. If you made these changes we would upstream them

**Comment by hanars** [comment]
I like how this looks better. I think what would make this a bit cleaner for you is you should make the `component` for the MOI field `LocusListItemsFilter` and make the `filterComponent` the custom `PaMoiSelector` component. This way, you get the subscription to the values change for the form locus and also the shared `shouldShow` behavior instead of needing to reimplement that. You could also update the `LocusListItemsFilter` to optionally pass a `locus` prop to all the child components so that way your MOI component can get it from there

**Comment by david-ma** [comment]
Hi Hana,

Sorry for the delay, I got a bit sick, but I’ve fixed the feature now, and I’ve merged seqr master into the feature branch to sort out the merge conflicts. The pull request [#2790](https://github.com/broadinstitute/seqr/pull/2790) is still open.

The changes:
PaMoiSelector has been moved from the header of the locus search panel to the body.
 PaMoiSelector is now a lot simpler and uses the FormSpy subscription from `LocusListItemsFilter` instead of trying to make its own. I’ve used the `panelAppItems` namespace (i.e. `PANEL_APP_FIELD_NAME`).
 PaMoiSelector is passed to LocusListItemsFilter as a filterComponent.

A `locus` prop is passed in from `LocusListItemsFilter`, so PaMoiSelector does not need to read the state.

The logic is contained in PaMoiSelector, and the changes to the `panelAppItems` fields are emitted to the other sibling components via formSpy's onChange(). Made possible because the `panelAppItems` namespace is used.

To avoid duplication of code, some shared logic for making the `panelAppItems` has been moved to `shared/utils/panelAppUtils.js`, as `panelAppLocusListReducer()`

Am I using FormSpy correctly here?

--

Also, Lodash has been removed, and the custom comparator logic from [Inputs.jsx](https://github.com/broadinstitute/seqr/pull/2790/files/b0a6093080be87fbdbda93b47b6ec24e801ce4c5#diff-ad459994f5ec2f6591fa9a3bf1ef66263a653718d5a5b4e6a9c0cd641553e144), has been extracted to `shared/utils/semanticUtils.js` so it can be reused whenever a deeper object comparator is needed than the default.
